### PR TITLE
[esomx] Remove undefined pins from esomx pinmap

### PR DESCRIPTION
**Pull request by scott-brust**
### Problem

There are defined pins that dont exist on the esomx hardware. These are causing failures when they are used, and masking problems when users use this pins and compliation succeeds. 

### Solution
Remove the pins from the pinmap, force explicit compilation failures to prompt fixing usage of the non existent pins.
This PR also fixes compliation problems with
- WIznet ethernet chip: If we want to use this with esomx we should reconsider the pin usage
- Tinker app: Specified the available pins for ESOMX hardware
- The 'User LED' for boron is on `D7` which doesnt exist on esomx. If we want to support this concept, I used `D5` instead
- The `wiring/api` test for Arduino expects a `LED_BUILTIN` define
- The `product_id_and_version` app expects a D7 pin to be available. 

### Steps to Test

Build for esomx platform

### Example App
use tinker

### References

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: hal/src/boron/network/network.cpp
Comment (avtolstoy) [inline hal/src/boron/network/network.cpp:128]: I'd propose we don't use pins that overlap with a peripheral, it is that way for other Gen 3 platforms, but shouldn't be the case here if avoidable.

File: hal/src/boron/pinmap_defines.h
Comment (avtolstoy) [inline hal/src/boron/pinmap_defines.h:198]: This doesn't look right
```
/* D21 / B3      - 17 */ { NRF_PORT_1,    13,          PIN_MODE_NONE, PF_NONE, ADC_CHANNEL_NONE, 2,                 3,                8, EXTI_CHANNEL_NONE, 0},
/* D24 / C0      - 18 */ { NRF_PORT_1,    8,           PIN_MODE_NONE, PF_NONE, ADC_CHANNEL_NONE, PWM_INSTANCE_NONE, PWM_CHANNEL_NONE, 8, EXTI_CHANNEL_NONE, 0},
```

File: user/applications/product_id_and_version/application.cpp
Comment (avtolstoy) [inline user/applications/product_id_and_version/application.cpp:9]: This should be defined for all platforms and this is an internal app. Can use `LED_PIN_USER` universally.

**Review by avtolstoy — CHANGES_REQUESTED** [review]

File: user/applications/product_id_and_version/application.cpp
Comment (scott-brust) [reply user/applications/product_id_and_version/application.cpp:9]: Ok, I didnt know if this define was available on gen2 platforms when we backport this PR to 4.x

Comment (avtolstoy) [reply user/applications/product_id_and_version/application.cpp:9]: Well, good news is that there is no Gen 2 support in 4.x :)

Comment (scott-brust) [reply user/applications/product_id_and_version/application.cpp:9]: oh, right 🙃

**Review by avtolstoy — APPROVED** [review]

**Review by technobly — APPROVED** [review]
LGTM!  After you merge to develop for 5.x, could you please make a new PR to target develop-4.x with the same commits?

File: hal/src/boron/pinmap_defines.h
Comment (technobly) [reply hal/src/boron/pinmap_defines.h]: D22/D23 is still missing in the pinmap_hal.c

File: hal/src/boron/network/network.cpp
Comment (technobly) [reply hal/src/boron/network/network.cpp]: Do we need to test this functionality with the new pins?

Comment (scott-brust) [reply hal/src/boron/network/network.cpp]: Ideally yes, but since there is no ethernet wing for the esomx it might take some extra effort to wire up. I'm not even sure this feature will be used for this platform.

**Pull request merged**
